ykcompile/src/lib.rs

```diff
@@ -125,8 +129,26 @@ impl TraceCompiler {
         }
     }
 
+    /// Returns the currently assigned register for a given `Local`. Similar to `local_to_reg` but
+    /// read-only, i.e. this function doesn't assign `Local`'s to registers.
+    fn get_reg(&self, local: &Local) -> Result<u8, CompileError> {
+        match self.assigned_regs.get(local) {
+            Some(u) => Ok(*u),
+            None => Err(CompileError::UnknownLocal),
+        }
+    }
+
     fn free_register(&mut self, local: &Local) {
         if let Some(reg) = self.assigned_regs.remove(local) {
+            if local == &self.rtn_var.as_ref().unwrap().local {
+                // We currently assume that we only trace a single function which leaves its return
```

Comment:
Gross hack, but OK to get us going ;)

Reply:
Indeed. It'll be a while before we won't need this workaround anymore.

Reply:
Ah! In fact I need this for my non-inline call work, which -- like for inlined calls -- puts the return value in the destination register, which may not be rax.